src/firewall/functions.py

```diff
@@ -101,6 +102,94 @@ def addr_family_bitsize(family):
 ###############################################################################
 
 
+def _parse_check_fcn(parse_fcn):
+    def f(*a, **kw):
+        try:
+            parse_fcn(*a, **kw)
+        except ValueError:
+            return False
+        return True
+
+    return f
+
+
+def _parse_norm_fcn(parse_fcn, unparse_fcn):
+    def f(*a, **kw):
+        detail = parse_fcn(*a, **kw)
+        return unparse_fcn(*detail)
+
+    return f
```

Comment:
These are only used by functions that I'm asking you to drop. Please also drop these.